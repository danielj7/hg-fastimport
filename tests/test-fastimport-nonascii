#!/bin/sh

# Test dumps with non-ASCII characters in various places.
#
# This file, including the fastimport dumps in it and the
# associated .out file, are all encoded in UTF-8.

. $TESTDIR/fastimport-common

echo "% import dump with non-ASCII committer"
cat > committer.dump <<__EOF__
commit refs/heads/master
mark :1
committer Jean-François <jf@example.com> 1007132290 +0000
data 4
foo

__EOF__

# N.B. it's import to run fastimport without overriding HGENCODING,
# because we need to be sure that UTF-8 metadata is correctly converted
# even when the default encoding is ASCII.  The catch is that the
# converted commit messages will be mangled as they are printed to
# stdout.  That's OK.  The real test is when we run hg log, and there we
# set HGENCODING to ensure that hg accurately prints what's in the
# repository.

hg init committer
cd committer
hg --traceback fastimport ../committer.dump
echo "% log with non-ASCII committer"
HGENCODING=utf-8 hg log
cd ..

echo "% import dump with non-ASCII author"
cat > author.dump <<__EOF__
commit refs/heads/master
mark :1
author Jürgen <juergen@example.org> 1250000000 +0400
committer Dave <dave@example.org> 1250000030 +0400
data 5
blah

__EOF__

hg init author
cd author
hg --traceback fastimport ../author.dump
echo "% log with non-ASCII author"
HGENCODING=utf-8 hg log
cd ..

echo "% import dump with non-ASCII message"
cat > message.dump <<__EOF__
commit refs/heads/master
mark :1
committer Dave <dave@example.org> 1250000030 +0400
data 60
fix naïve implementation that Øle threw together for €5

M 644 inline foo.txt
data 6
hello


__EOF__

hg init message
cd message
hg --traceback fastimport ../message.dump | sed "s|$HGTMP|HGTMP|g"
echo "% log with non-ASCII message"
HGENCODING=utf-8 hg log
echo "% manifest"
hg manifest -r0
echo "% contents of foo.txt"
hg cat -r0 foo.txt
cd ..
