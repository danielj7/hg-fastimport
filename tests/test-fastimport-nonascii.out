% import dump with non-ASCII committer
scanning source...
sorting...
converting...
0 foo
% log with non-ASCII committer
changeset:   0:f1709065901b
tag:         tip
user:        Jean-François <jf@example.com>
date:        Fri Nov 30 14:58:10 2001 +0000
summary:     foo

% import dump with non-ASCII author
scanning source...
sorting...
converting...
0 blah
% log with non-ASCII author
changeset:   0:639a4e09e8af
tag:         tip
user:        Jürgen <juergen@example.org>
date:        Tue Aug 11 18:13:20 2009 +0400
summary:     blah

% import dump with non-ASCII message
scanning source...
sorting...
converting...
0 fix na?ve implementation that ?le threw together for ?5
Removing blob dir 'HGTMP/test-fastimport-nonascii/message/.hg/blobs' ...
% log with non-ASCII message
changeset:   0:52b3afdff0e7
tag:         tip
user:        Dave <dave@example.org>
date:        Tue Aug 11 18:13:50 2009 +0400
summary:     fix naïve implementation that Øle threw together for €5

% manifest
foo.txt
% contents of foo.txt
hello
